pyproject.toml

```diff
@@ -41,8 +41,7 @@ autoflake = "^2.2.1"
 isort = "^5.13.2"
 markdownify = "^0.11.6"
 tavily-python = "^0.3.1"
-microchain = "^0.0.4"
-microchain-python = "^0.3.4"
+microchain-python = {git = "https://github.com/evangriffiths/microchain.git", rev = "evan/token-tracker"}
```

Comment:
I would just add a todo to remind ourselves to revert to main once the PR is accepted. Otherwise looks good.